pkg/storegateway/bucket.go

```diff
@@ -1139,6 +1139,13 @@ func (s *BucketStore) getSeriesIteratorFromBlocks(
 			return nil
 		})
 
+		if b.meta.Compaction.Level == 1 && b.meta.Thanos.Source == block.ReceiveSource && !b.queriedNonCompactedBlock {
```

Comment:
Since this is debug logging and will be off most of the time, I'd get rid of the of `queriedNonCompactedBlock` field and log every time it's queried or every N times it's queried instead of mutating the block.

Reply:
I did consider that, and I agree with your comment above about `queriedNonCompactedBlock` being odd, but even with a `Sampler`, we could have a lot of log entries. When testing this without any sampling, I occasionally saw millions of log entries in a relatively small time period. So it might be hard to configure a sampler accurately - where either we don't log enough or we log too much. A rate limiter might be easier to configure, but we'd still miss some log IDs with that approach. Ultimately, logging each block ID just once is ideal, which is how I ended up there. That said, I agree it is odd.

Reply:
another option is to log it every time it is loaded - that means on startup or when the block is discovered while the store-gateway is running. The store-gateway filters blocks which are too young. So loading the block should mean it's effectively being queried, right? My assumption is that uncompacted blocks are recent blocks are they are almost constantly queried, this is less valid if the uncompacted block is for an older time range (days, weeks).

Reply:
That's a good suggestion, and yea, any block that the store gateway creates a reader for has already been filtered based on a query. I updated the PR to do this.

Reply:
I had more `addBlock` in mind https://github.com/grafana/mimir/blob/08ef78f1549966459745aa094200c955dd1ef5c0/pkg/storegateway/bucket.go#L398-L410

That would also remove the need to track the boolean in the `block` struct, which I think is what Nick was alluding to

Reply:
`addBlock` gets called for all blocks discovered in the bucket. The filtering comes after that, [here](https://github.com/grafana/mimir/blob/08ef78f1549966459745aa094200c955dd1ef5c0/pkg/storegateway/bucket.go#L599), at query time.